iOSDFULibrary/Classes/Implementation/SecureDFU/Characteristics/SecureDFUPacket.swift

```diff
@@ -59,7 +59,7 @@ internal class SecureDFUPacket: DFUCharacteristic {
         
         if #available(iOS 9.0, macOS 10.12, *) {
             // Level changes to iOS DFU framework: Make the packet size the first word-aligned value that's less than the maximum
-            packetSize = UInt32(characteristic.service.peripheral.maximumWriteValueLength(for: .withoutResponse)) & 0xFFFC
+            packetSize = UInt32(characteristic.service?.peripheral?.maximumWriteValueLength(for: .withoutResponse) ?? 0) & 0xFFFC
```

Comment:
I'm not sure we should use `0` here. Will this work?
```suggestion
        if #available(iOS 9.0, macOS 10.12, *), let peripheral = characteristic.service?.peripheral {
            // Level changes to iOS DFU framework: Make the packet size the first word-aligned value that's less than the maximum
            packetSize = UInt32(peripheral.maximumWriteValueLength(for: .withoutResponse)) & 0xFFFC
```